src/core/send.c

```diff
@@ -317,7 +327,8 @@ BOOLEAN
 QuicSendSetStreamSendFlag(
     _In_ QUIC_SEND* Send,
     _In_ QUIC_STREAM* Stream,
-    _In_ uint32_t SendFlags
+    _In_ uint32_t SendFlags,
+    _In_ BOOLEAN DelaySend
```

Comment:
why does this need to be a separate field instead of just another `SendFlag`?

Reply:
This isn't a flag because flags represent something to send. The design here is to queue all things that need to be sent, but don't actually flush the data yet. If something else results in a flush, go ahead and process all the flags then.

Reply:
Ok, so this means that if something else needs to be sent (ACK, Ping, etc.) it won't be delayed by stream data that is delayed?

Reply:
Notice this function is stream specific. Nothing connection-wide goes through here, so those definitely trigger an immediate flush. Non-data stream frames will also trigger immediate flush too.